sigmahq title first draft (#95)
SigmaHQ/sigma-specification: sigmahq/sigmahq_title_rule.md

## Structure

Titles can be split with "-" : `Prefix - Main Title - Sufix`

### Prefix (Optional)

It is used to give a category, type of malware or name a threat actor. The choice depends highly on the type of rule.

Examples:

- HackTool
- PUA
- Remote Access Tool

Specific wording example:
- "ATP27 - "
- "ATP29 - "
- "UNC2452 - "
- "UNC4841 - "

### Suffix (Optional)

Sometimes the detections are duplicated across different `logsource`s with little changes to their logic. This is common in the case of Process Creation rules targeting the PowerShell process. Those rules are typically duplicated for the different PowerShell `logsource`s using ScriptBlockText to check for the same characteristics. A suffix in this case will be used to differentiate between the rules of the different `logsource`s.

Example:

```yaml
title: Invoke-Obfuscation Obfuscated IEX Invocation
title: Invoke-Obfuscation Obfuscated IEX Invocation - PowerShell
title: Invoke-Obfuscation Obfuscated IEX Invocation - PowerShell Module
title: Invoke-Obfuscation Obfuscated IEX Invocation - Security
title: Invoke-Obfuscation Obfuscated IEX Invocation - System
```

> ### Description

Reviewer: ```suggestion
### Main Title

```

# SigmaHQ Rule Conventions <!-- omit in toc -->

This document describes how to write a title for a sigma rule.
It is impossible to be exhaustive, so it is more of a guide than a standard.

## Summary

- [Summary](#summary)
- [Generality](#generality)
- [Structure](#structure)
  - [Prefix](#prefix)
  - [Suffix](#suffix)
  - [Description](#description)

## Generality

Bear in mind that the title is the first thing the operator will see.
It should therefore be used as a clue to guide the consideration of the alert.

The title and level of the rule must be consistent

## Structure

> Title can be split with "-" : `Prefix - Description - Sufix`

Reviewer 2: ```suggestion
Titles can be split with "-" : `Prefix - Description - Sufix`

```

### Prefix
It is Optional.
It is used to give a category or type of malware, an APT 

> Commun wordding:

Reviewer 2: ```suggestion
Examples:

```

> - Hack Tool
> - Lolbin (lolbas)
> - PUA
> - Remote Access Tool

Reviewer 2: ```suggestion
- HackTool
- PUA
- Remote Access Tool

```

Specific wording example:
- "ATP27 - "
- "ATP29 - "
- "UNC2452 - "
- "UNC4841 - "

> ### Suffix
> It is Optional.

Reviewer 2: ```suggestion
### Suffix (Optional)

```

> It is used to differentiate the same detection but on a different logsource.

Reviewer 2: ```suggestion
Sometimes the detection are duplicated across different log-source with little changes to their logic. This is common in the case of Process Creation rules targeting the PowerShell process and rules using ScriptBlockText to check for the same. A suffix in this case will be used to offer such distinction.

```

> Exemple:

Reviewer 2: ```suggestion
Example:

```

```yaml
title: Invoke-Obfuscation Obfuscated IEX Invocation
title: Invoke-Obfuscation Obfuscated IEX Invocation - PowerShell
title: Invoke-Obfuscation Obfuscated IEX Invocation - PowerShell Module
title: Invoke-Obfuscation Obfuscated IEX Invocation - Security
title: Invoke-Obfuscation Obfuscated IEX Invocation - System
```

### Description

The point here is to explain the alert in a meaningful way.
The title does not need to use the terms "Detect", "Detection" or "Detection".

We use a simple formula to describe the alert.
Example:
- "7Zip Compressing ..."
- "Add User to ..."
- "Bypass UAC Using ..."
- "Renamed xxx Execution"
- "UAC Bypass Using ..."

`informational` and `low` are not intended for everyday use and are subject to many false positives.
The title should therefore be general.

Example : `Net.exe Execution` 

`medium` rules can have false positives and requires further analysis
Wording :

> - "Suspicious "

Reviewer 2: ```suggestion
- "Potential "

```

`high` rules requires a prompt review
Wording :

> - "Potential "

Reviewer 2: ```suggestion
- "Suspicious "

```

The title and level of the rule must be consistent

## Structure

Titles can be split with "-" : `Prefix - Description - Sufix`

### Prefix (Optional)

It is used to give a category or type of malware, an APT 

Examples:

- HackTool
- PUA
- Remote Access Tool

Specific wording example:
- "ATP27 - "
- "ATP29 - "
- "UNC2452 - "
- "UNC4841 - "

### Suffix (Optional)

Sometimes the detection are duplicated across different log-source with little changes to their logic. This is common in the case of Process Creation rules targeting the PowerShell process and rules using ScriptBlockText to check for the same. A suffix in this case will be used to offer such distinction.

Example:

```yaml
title: Invoke-Obfuscation Obfuscated IEX Invocation
title: Invoke-Obfuscation Obfuscated IEX Invocation - PowerShell
title: Invoke-Obfuscation Obfuscated IEX Invocation - PowerShell Module
title: Invoke-Obfuscation Obfuscated IEX Invocation - Security
title: Invoke-Obfuscation Obfuscated IEX Invocation - System
```

### Description

The point here is to explain the alert in a meaningful way.

> The title does not need to use the terms "Detect", "Detection" or "Detection".

Reviewer 2: The description here refers to the structure defined above `Titles can be split with "-" : Prefix - Description - Sufix`

Maybe we can find a better word than description to not get it confused with the "real" description field.

> The title does not need to use the terms "Detect", "Detection" or "Detection".

Author: `Titles can be split with "-" : Prefix - Main Text - Sufix` ?

We use a simple formula to describe the alert.
Example:
- "7Zip Compressing ..."
- "Add User to ..."
- "Bypass UAC Using ..."
- "Renamed xxx Execution"
- "UAC Bypass Using ..."

`informational` and `low` are not intended for everyday use and are subject to many false positives.
The title should therefore be general.

Example : `Net.exe Execution` 

> `medium` rules can have false positives and requires further analysis

Reviewer: ```suggestion
`medium` rules can have environment dependent false positives and require a tuning/evaluation phase before deploying to production environments.

```

Wording :
- "Potential "

`high` rules requires a prompt review
Wording :
- "Suspicious "

> `critical` rules should be reviewed immediately

Reviewer: ```suggestion
`critical` rules should be reviewed with the highest priority.

```

> `critical` rules should be reviewed immediately
> The title must therefore be precise.

Reviewer: maybe? 
```suggestion
`critical` rules should be reviewed immediately
The title must therefore be precise and indicate the threat.

Keywords used to indicate this:
- "Malware"
- "Exploit"
- "... Attempt"
- "<Threat Actor> Activity"
```

> The title must therefore be precise.

Reviewer: But as I said. I wouldn't want to enforce anything there. First we would need to find good keywords to use.

> The title must therefore be precise.

Author: I wanted to remain as open as possible.
From a syntax perspective "Use of 7zip" and "Use of Brute Ratel" are the same.
Operator have the title and the level to differentiate them.

 "High-Risk Use of Brute Ratel"  , but why high for a critical 😄 
threatening, vicious, menacing ...

I don't see any solution

> The title must therefore be precise.

Reviewer 2: I think the intention is clear, we don't enforce it to not be cumbersome to the author but we'll review anyway so just having it as its now is good guidance, especially when the author will look at some example of critical rules. They should get the idea.

> The title must therefore be precise.

Reviewer: What's with the code suggestion of this thread? I think it makes sense to give examples, doesn't it?

> The title must therefore be precise.

Reviewer 2: Applied the suggestion
